src/libfsm/vm/v1.c

```diff
@@ -411,7 +411,11 @@ vm_match_v1(const struct dfavm_v1 *vm, struct vm_state *st, const char *buf, siz
 					rel = packed.i;
 				}
 
-				st->pc += rel;
+				if (rel >= 0) {
+					st->pc += (uint32_t)rel;
+				} else {
+					st->pc -= (uint32_t)-rel;
+				}
```

Comment:
Sorry to be late to comment here, but I think there's a simpler change that avoids adding a branch.  `rel` is an `int32_t` and `st->pc` is a `uint32_t`.  We should be able to cast `rel` to `uint32_t` and just add them; the unsigned wrap-around will take care of the rest:
```
st->pc += (uint32_t)rel;
```